### PR TITLE
Fix the incorrect width that occurred when formatting a floating number in the locale specific form

### LINKED ISSUE #4316: `<format>`: The width of output is miscalculated when formatting a floating-point number in the locale-specific form
# Describe the bug
Revealed by libc++ test `std/utilities/format/format.functions/locale-specific_form.pass.cpp`.

When formatting a floating-point number, digit separators are not taken into account when computing the width of output, which may cause too many fill characters in the output.

For integers, digit separators are correctly handled:

https://github.com/microsoft/STL/blob/3eac329d1f614ecf138d96c22a3b02f87076bc4a/stl/inc/format#L2873-L2878

For floating-point numbers, it seems that the width is not adjusted:

https://github.com/microsoft/STL/blob/3eac329d1f614ecf138d96c22a3b02f87076bc4a/stl/inc/format#L3099-L3102

# Command-line test case
```
D:\test>type test-format.cpp
#include <format>
#include <iostream>

int main() {
    std::cout << std::format(std::locale("en_US"), "{:@>8L}\n", 12345);
    std::cout << std::format(std::locale("en_US"), "{:@>8L}\n", 12345.0);
}

D:\test>cl /std:c++20 /utf-8 /EHs test-format.cpp
Microsoft (R) C/C++ Optimizing Compiler Version 19.39.33321 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

test-format.cpp
Microsoft (R) Incremental Linker Version 14.39.33321.0
Copyright (C) Microsoft Corporation.  All rights reserved.

/out:test-format.exe
test-format.obj

D:\test>test-format.exe
@@12,345
@@@12,345
```

# Expected behavior
The output should be `@@12,345` instead of `@@@12,345`.

# STL version

    https://github.com/microsoft/STL/commit/3eac329d1f614ecf138d96c22a3b02f87076bc4a

### PR DESCRIPTION
**Pull request by grcm10**
Fixes #4316

### CODE REVIEW

**Comment by frederick-vs-ja** [comment]
These lines need to be removed:

https://github.com/microsoft/STL/blob/c53ac59abae552e50654943f7f80f23376a8039a/tests/libcxx/expected_results.txt#L571-L574

File: tests/std/tests/P0645R10_text_formatting_formatting/test.cpp [hunk 1]
```diff
@@ -1506,6 +1506,12 @@ constexpr bool test_format_string() {
     return true;
 }
 
+template <class charT>
+void test_gh_4316() {
```
Comment (StephanTLavavej) [inline tests/std/tests/P0645R10_text_formatting_formatting/test.cpp:1510]: Like below, let's capture the bug title here, to help readers understand what this is testing.

**Review by StephanTLavavej — APPROVED** [review]

File: tests/std/tests/P0645R10_text_formatting_formatting/test.cpp (hunk 1 above)
Comment (grcm10) [reply tests/std/tests/P0645R10_text_formatting_formatting/test.cpp]: Thanks for reminder!

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks for fixing this runtime correctness bug and unblocking more libcxx tests! :lady_beetle: :zap: :heart_eyes_cat:

**Pull request merged**